### ISSUE TITLE
CodiMD keeps reloading

### ISSUE THREAD

**Issue by JayDee2202** [hackmdio/codimd#1005]
As wished [here](https://github.com/hackmdio/codimd/issues/727#issuecomment-428645698) I open a new issue for that error pattern.

After updating to version 1.2.0 I received errors from the wekan app in the backend and the frontend.
Everything works, but the message is still there, even after updating to 1.2.1. If I click on "reload" at the dialog shown the site reloads and the message reappears. By clicking the X everything works normally at the frontend.

![chrome_2018-10-10_17-53-31](https://user-images.githubusercontent.com/15029826/46753523-6845eb80-ccc0-11e8-89c8-243596e0269e.png)

I'm using docker with postgres sql and update via the container.
Error message from the wekan-app:

```
2018-10-10T15:44:27.406Z - error:  TypeError: Cannot read property 'charAt' of undefined
    at Object._decompress (/hackmd/node_modules/lz-string/libs/lz-string.js:473:25)
    at Object.decompressFromBase64 (/hackmd/node_modules/lz-string/libs/lz-string.js:44:21)
    at Model.<anonymous> (/hackmd/lib/history.js:35:29)
    at Model.tryCatcher (/hackmd/node_modules/bluebird/js/release/util.js:16:23)
    at Promise._settlePromiseFromHandler (/hackmd/node_modules/bluebird/js/release/promise.js:512:31)
    at Promise._settlePromise (/hackmd/node_modules/bluebird/js/release/promise.js:569:18)
    at Promise._settlePromise0 (/hackmd/node_modules/bluebird/js/release/promise.js:614:10)
    at Promise._settlePromises (/hackmd/node_modules/bluebird/js/release/promise.js:693:18)
    at Async._drainQueue (/hackmd/node_modules/bluebird/js/release/async.js:133:16)
    at Async._drainQueues (/hackmd/node_modules/bluebird/js/release/async.js:143:10)
    at Immediate.Async.drainQueues (/hackmd/node_modules/bluebird/js/release/async.js:17:14)
    at runCallback (timers.js:810:20)
    at tryOnImmediate (timers.js:768:5)
    at processImmediate [as _immediateCallback] (timers.js:745:5)
```

So I guess some step of the database migration failes somewhere.
I update via `docker-compose pull && docker-compose up` and I'm using nearly the provided compose-file:

```
# Using version 3 to provide play-with-docker badge
# You can change to version 2 without breaking.
#version: '2'
version: '3'
services:
  database:
    # Don't upgrade postgres by simply changing the version number
    # You need to migrate the Database to the new postgres version
    image: postgres:9.6-alpine
    container_name: hackmd-db
    environment:
      - POSTGRES_USER=hackmd
      - POSTGRES_PASSWORD=<SECRETPASSWORD>
      - POSTGRES_DB=hackmd
    volumes:
      - /data/hackmd/:/var/lib/postgresql/data
    networks:
      backend:
        aliases:
          - hackmdPostgres
          - hackmdpostgres
    restart: always
  app:
    # Uncomment the following section to build the image yourself:
    #build:
    #  context: .
    #  dockerfile: debian/Dockerfile
    #  args:
    #    - "VERSION=master"
    #    - "HACKMD_REPOSITORY=https://github.com/hackmdio/hackmd.git"
    image: hackmdio/hackmd:1.2.0
    container_name: hackmd-app
    environment:
      # DB_URL is formatted like: <databasetype>://<username>:<password>@<hostname>/<database>
      # Other examples are:
      # - mysql://database:3306/database
      # - sqlite:///data/sqlite.db (NOT RECOMMENDED)
      # - For details see the official sequelize docs: http://docs.sequelizejs.com/en/v3/
      - CMD_DB_URL=postgres://hackmd:<SECRETPASSWORD>@hackmdPostgres:5432/hackmd
      - CMD_ALLOW_FREEURL=true
    ports:
      # Ports that are published to the outside.
      # The latter port is the port inside the container. It should always stay on 3000
      # If you only specify a port it'll published on all interfaces. If you want to use a
      # local reverse proxy, you may want to listen on 127.0.0.1.
      # Example:
      # - "127.0.0.1:3000:3000"
      - "3000:3000"
    networks:
      backend:
    restart: always
# Define networks to allow best isolation
networks:
  # Internal network for communication with Postgres
  backend:

# Define named volumes so data stays in place
volumes:
  # Volume for Postgres database
  database:
```

I've currently no idea where this came from and I didn't miss a major update. Version 1.0.1-ce worked properly and the 1.1.0, too.
Is there something I missed doing?

**Comment by SISheogorath** [comment]
Do you run it behind a reverse proxy?

**Comment by JayDee2202** [comment]
Yes, it does, at port 3000.

**Comment by SISheogorath** [comment]
I guess the yes is about the reverse proxy?

> And if so, do you set a nosniff option for the content type?
This is still an outstanding question.

And I would also add: Does your reverse proxy does caching?

**Comment by JayDee2202** [comment]
Sorry, I already sent the answer and didn't saw the edited question.

Yes, I'm using a reverse proxy and I didn't set a nosniff option. It doesn't cache, it's working via apache 2.4.25 and ProxyPass/ProxyPassReverse via vHost settings.

**Comment by SISheogorath** [comment]
Check what's the response from your instance's `/config` path using the browser's dev tools?

And then check it inside the container using `docker-compose exec app wget -O- http://127.0.0.1:3000/config`. 

Do the results match?
Does the browser use port 3000? If so, set `CMD_DOMAIN`, `CMD_PROTOCOL_USESSL` and `CMD_URL_ADDPORT`.

Apart from that you should consider to use the suggestion from the port section when your reverse proxy runs on the same host as your CodiMD instance.

**Comment by JayDee2202** [comment]
Yes, the results match.

I'm using a subdomain for the proxy, so the browser connects to the subdomain and is redirected to the HTTPS subdomain. This connects via reverse proxy to the container, so the user didn't notice anything about running CodiMD in a container.

After you named the config, I just tested connecting directly to IP and port without domain, TLS, proxy etc. and everything works fine. No notification appeared.
But even with setting `CMD_DOMAIN`, `CMD_PROTOCOL_USESSL` and `CMD_URL_ADDPORT` in the compose file it still doesn't work via subdomain. I didn't modify the subdomain since I set that containers up, the issues are appearing since updating to version 1.2.0 so I think there must have changed something. It might conflict with the current proxy settings, I think.

The proxy and the container runs at the same machine, yes.

**Comment by SISheogorath** [comment]
Yes, there was a change in version 1.2.0 and a bug that triggered this behaviour. But it was fixed in 1.2.1, so please do all your tests on 1.2.1.

If it still appears, please provide your reverse proxy config along with the steps to reproduce the issue, otherwise there is no way to really find the issue without wild guessing, which I would like to avoid.

**Comment by bonanza123** [comment]
I had the same problem with similar setup and it seems to solved with 1.2.1. Thanks!

**Comment by JayDee2202** [comment]
Okay, it seems, the container didn't build properly. I already chanced the tag to 1.2.1 but the error remains, after removing the containers, deleting the image and pulling everything again it works.

Thanks so far for the helpful and quick replys!
